run_mcmc.cc

```diff
@@ -22,7 +22,7 @@ bool check_args(argparse::ArgumentParser args) {
   if (args["--coll-name"] != std::string("memory")) {
     check_file_is_writeable(args.get<std::string>("--coll-name"));
   }
-  if (args["--grid-file"] != std::string("\"\"")) {
+  if (args["--dens-file"] != std::string("\"\"")) {
```

Comment:
this was intended: if "grid-file" is provided, we need to write to "dens-file". Maybe we could handle better the situation where grid-file is provided but dens-file is not and/or vice-versa?

Reply:
To me it makes sense to check an argument if _it_, not another argument, is non-null. Plus the previous version of the code was writing to the file named `""` when one of them was not provided. We should only write to the density if both `grid-file` and `dens-file` are non-empty, which is what my updated version should do

Reply:
excellent, my bad then! Please merge at will